nixos/modules/services/networking/wireguard.nix

```diff
@@ -134,15 +162,16 @@ let
 
   };
 
-  generateConf = name: values: pkgs.writeText "wireguard-${name}.conf" ''
+  generateConf = name: values: ''
     [Interface]
-    PrivateKey = ${values.privateKey}
+    PrivateKey = ${if values.privateKeyFile != null then "$(cat ${values.privateKeyFile})" else values.privateKey}
     ${optionalString (values.listenPort != null)   "ListenPort = ${toString values.listenPort}"}
 
     ${concatStringsSep "\n\n" (map (peer: ''
     [Peer]
     PublicKey = ${peer.publicKey}
-    ${optionalString (peer.presharedKey != null) "PresharedKey = ${peer.presharedKey}"}
+    ${optionalString (peer.presharedKeyFile != null) "PresharedKey = $(cat ${peer.presharedKeyFile})"}
+    ${optionalString (peer.presharedKey != null && peer.presharedKeyFile == null) "PresharedKey = ${peer.presharedKey}"}
```

Comment:
This logic here is different and more complex than the logic above for privatekey.